shedskin/cpp.py

```diff
@@ -599,45 +612,48 @@ def do_comments(self, child):
             for n in self.gx.comments[child]:
                 self.do_comment(n)
 
-    def visitContinue(self, node, func=None):
+    def visit_Continue(self, node, func=None):
```

Comment:
Could you please write all lowercase when using `_`, in this case `visit_continue`?
This matches the naming conventions in the other modules.

Reply:
> Could you please write all lowercase when using _, in this case visit_continue?
This matches the naming conventions in the other modules.

Here it's not about naming conventions in the other modules, here's it's `visit_` + name of an `ast` module *class*. And majority of ast module classes are upper-case, by Python conventions (also to avoid need for "quoting", e.g. class `While` is possible, but `while` isn't; oh and yeah, some classes in `ast` are lower-case ;-). Mostly for aux, constituent constructs, like `ast.comprehension`, where `ast.comprehension` is factored out part of specific list vs set vs dict vs genexp comprehensions).

Reply:
These names are conforming to the ast module NodeVisitor implementation, as @pfalcon says the capitalization follows from the class names.

I do think it makes sense for these functions to break the naming convention, signaling they are special. I've added the 'impl_' prefix to some functions  named 'visit_' to distinguish these from the visitor pattern functions, but it turns out i've missed `visit_tuple_or_list`, `visit_callfunc_args` in cpp.py and `visit_impl_bitop` in graph.py